libbeat/common/worker.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"sync"
+)
+
+type WorkerSignal struct {
+	Done    chan struct{}
```

Comment:
oh right, my fault. Another way preventing potential sends is having 'Done' private and a method returning "<- chan struct{}".

All in all, I'm fine living with convention "never send or close Done but use for signaling 'closed' only". Seems to be a quite common convention, no need to any additional boilerplate.